src/pkg/otelcolclient/otelcolclient.go

```diff
@@ -131,6 +153,76 @@ func (c *Client) Close() error {
 	return c.b.w.Close()
 }
 
+// writeCounter translates a loggregator v2 Counter to OTLP and adds the metric to the pending batch.
+func (c *Client) writeLog(e *loggregator_v2.Envelope) {
+	atts := attributes(e)
+	svrtyNumber := logspb.SeverityNumber_SEVERITY_NUMBER_UNSPECIFIED
+	switch e.GetLog().GetType() {
```

Comment:
LOG_OUT and LOG_ERR only show if the log message was collected from STDOUT or STDERR, but they don't show the log level.

The log level can be something very different than info and error and Otel [defines](https://github.com/open-telemetry/opentelemetry-collector/blob/main/pdata/internal/data/protogen/logs/v1/logs.pb.go#L35-L62) many and very granular log levels.

Here is an example app:
```go
package main

import (
	"log/slog"
	"os"
)

func main() {
	stdoutLogger := slog.New(slog.NewTextHandler(os.Stdout, &slog.HandlerOptions{Level: slog.LevelDebug}))
	stderrLogger := slog.New(slog.NewTextHandler(os.Stderr, &slog.HandlerOptions{Level: slog.LevelDebug}))

	slog.SetDefault(stdoutLogger)
	slog.Debug("stdout debug message")
	slog.Info("stdout info message")
	slog.Warn("stdout warn message")
	slog.Error("stdout error message")
	slog.SetDefault(stderrLogger)
	slog.Debug("stderr debug message")
	slog.Info("stderr info message")
	slog.Warn("stderr warn message")
	slog.Error("stderr error message")
}
```

For which the output is something like:
```bash
cf logs <app_name>
2024-07-17T13:24:27.93+0000 [APP/PROC/WEB/0] OUT time=2024-07-17T13:24:27.931Z level=DEBUG msg="stdout debug message"
2024-07-17T13:24:27.93+0000 [APP/PROC/WEB/0] OUT time=2024-07-17T13:24:27.932Z level=INFO msg="stdout info message"
2024-07-17T13:24:27.93+0000 [APP/PROC/WEB/0] OUT time=2024-07-17T13:24:27.932Z level=WARN msg="stdout warn message"
2024-07-17T13:24:27.93+0000 [APP/PROC/WEB/0] OUT time=2024-07-17T13:24:27.932Z level=ERROR msg="stdout error message"
2024-07-17T13:24:27.93+0000 [APP/PROC/WEB/0] ERR time=2024-07-17T13:24:27.932Z level=DEBUG msg="stderr debug message"
2024-07-17T13:24:27.93+0000 [APP/PROC/WEB/0] ERR time=2024-07-17T13:24:27.932Z level=INFO msg="stderr info message"
2024-07-17T13:24:27.93+0000 [APP/PROC/WEB/0] ERR time=2024-07-17T13:24:27.932Z level=WARN msg="stderr warn message"
2024-07-17T13:24:27.93+0000 [APP/PROC/WEB/0] ERR time=2024-07-17T13:24:27.932Z level=ERROR msg="stderr error message"
```

As it can be seen, depending on the logger used in the app and its config, the log level can be something very different than info and error. If we want to set the log level properly we have to check the contents of the message.

@cloudfoundry/wg-app-runtime-platform-logging-and-metrics-approvers Any thoughts on this?

Reply:
The reason to map the severity from the [envelope Log_Type is because there are only 2 options](https://pkg.go.dev/code.cloudfoundry.org/go-loggregator/rpc/loggregator_v2#Log_Type). At this stage -converting from a v2 envelope to OTLP- "there is no way" (well, let's say, it is not convenient) to parse the log payload to define the real log level. That can be done later with a transform processor using OTTL (see https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/transformprocessor/README.md#get-severity-of-an-unstructured-log-body).

Regarding this topic, we have committed the same logic to the [cloudfoundryreceiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/cloudfoundryreceiver):  https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/cloudfoundryreceiver/converter.go#L52-L58 .  Users/operators are free to set the correct level with additional transforms, or adding additional attributes. 

A bit of context for this decision is because we are using Grafana Loki and visualizing the logs in Grafana automatically sets the colors of the logs in the panel depending on the severity level:  green for "INFO", red for "ERROR", gray for unknown.

Reply:
This approach is similar to how the Syslog Agent figures out severity for log messages: https://github.com/cloudfoundry/loggregator-agent-release/blob/04137edd129c85879e778730adbcd8a3afd1d5aa/src/pkg/egress/syslog/rfc5424.go#L150.

Given the precedent, I'm onboard with this approach for now.

Reply:
Hm.... good catch @ctlong! :) I missed that :-/ 

In that case, we can set this to have some defaults and I guess let the app developer or cf operator, to be able to add a processor and re-adjust things if needed.